packages/core/types/src/common/config-module.ts

```diff
@@ -590,6 +593,42 @@ export type ProjectConfigOptions = {
    * ```
    */
   worker_mode?: "shared" | "worker" | "server"
+
+  /**
+   * Configure the application's authentication settings
+   *
+   * @example
+   * ```js title="medusa-config.js"
+   * module.exports = {
+   *   projectConfig: {
+   *     auth: {
+   *       jwtSecret: "some-super-secret",
+   *       jwtExpiresIn: "2d",
+   *     }
+   *     // ...
+   *   },
+   *   // ...
+   * }
+   * ```
+   */
+  auth: {
+    /**
+     * The secret to sign the JWT token, usually a randomly generated string. If not provided, the default value is `supersecret`.
+     */
+    jwtSecret: string
+    /**
+     * The expiration time for the JWT token. If not provided, the default value is `24h`.
+     */
+    jwtExpiresIn: string
+    /**
+     * The secret to sign the cookie session, usually a randomly generated string. If not provided, the default value is `supersecret`.
+     */
+    cookieSecret: string
+    /**
+     * Specify the accepted URLs or patterns for API Routes starting with `/auth`. It can either be one accepted origin, or a comma-separated list of accepted origins.
+     */
+    cors: string
```

Comment:
Q: can't it be a regexp like string? like for the admin/store cors, in the comment it looks like it is still the old description

Reply:
Yup, haven't added the full JS comment yet. Wanted approval first 👍